pymongo/common.py

```diff
@@ -524,56 +525,77 @@ def validate_tzinfo(dummy, value):
+    'compressors': validate_compressors,
+    'connecttimeoutms': validate_timeout_or_none,
+    'heartbeatfrequencyms': validate_timeout_or_none,
     'journal': validate_boolean_or_string,
+    'localthresholdms': validate_positive_float_or_zero,
+    'maxidletimems': validate_timeout_or_none,
     'maxpoolsize': validate_positive_integer_or_none,
-    'socketkeepalive': validate_boolean_or_string,
-    'waitqueuemultiple': validate_non_negative_integer_or_none,
-    'ssl': validate_boolean_or_string,
-    'ssl_keyfile': validate_readable,
-    'ssl_certfile': validate_readable,
-    'ssl_pem_passphrase': validate_string_or_none,
-    'ssl_cert_reqs': validate_cert_reqs,
-    'ssl_ca_certs': validate_readable,
-    'ssl_match_hostname': validate_boolean_or_string,
-    'ssl_crlfile': validate_readable,
+    'maxstalenessseconds': validate_max_staleness,
     'readconcernlevel': validate_string_or_none,
     'readpreference': validate_read_preference_mode,
     'readpreferencetags': validate_read_preference_tags,
-    'localthresholdms': validate_positive_float_or_zero,
-    'authmechanism': validate_auth_mechanism,
-    'authsource': validate_string,
-    'authmechanismproperties': validate_auth_mechanism_properties,
-    'tz_aware': validate_boolean_or_string,
-    'uuidrepresentation': validate_uuid_representation,
-    'connect': validate_boolean_or_string,
-    'minpoolsize': validate_non_negative_integer,
-    'appname': validate_appname_or_none,
-    'driver': validate_driver_or_none,
-    'unicode_decode_error_handler': validate_unicode_decode_error_handler,
+    'replicaset': validate_string_or_none,
     'retrywrites': validate_boolean_or_string,
-    'compressors': validate_compressors,
+    'serverselectiontimeoutms': validate_timeout_or_zero,
+    'sockettimeoutms': validate_timeout_or_none,
+    'tls': validate_boolean_or_string,
+    'tlsallowinvalidcertificates': validate_allow_invalid_certs,
+    'ssl_cert_reqs': validate_cert_reqs,
+    'tlsallowinvalidhostnames': validate_boolean_or_string,
+    'tlscafile': validate_readable,
+    'tlsclientcertfile': validate_readable,
+    'tlsclientkeypassword': validate_string_or_none,
+    'tlsclientkeyfile': validate_readable,
+    'w': validate_non_negative_int_or_basestring,
+    'wtimeoutms': validate_non_negative_integer,
     'zlibcompressionlevel': validate_zlib_compression_level,
 }
 
-TIMEOUT_VALIDATORS = {
-    'connecttimeoutms': validate_timeout_or_none,
-    'sockettimeoutms': validate_timeout_or_none,
+# Dictionary where keys are the names of URI options specific to pymongo,
+# and values are functions that validate user-input values for those options.
+NONSPEC_OPTIONS_VALIDATOR_MAP = {
+    'connect': validate_boolean_or_string,
+    'driver': validate_driver_or_none,
+    'fsync': validate_boolean_or_string,
+    'minpoolsize': validate_non_negative_integer,
+    'socketkeepalive': validate_boolean_or_string,
+    'tlscrlfile': validate_readable,
```

Comment:
Surprising that `tlscrlfile` is not part of the new spec. Why is that?

Reply:
Only the Python driver exposes a CRL file feature. None of the other drivers have bothered to do it and as per a discussion I had with Sam Rossi, there doesn't seem to be much appetite to start now.